pylint/lint/pylinter.py

```diff
@@ -1068,16 +1119,19 @@ def _check_astroid_module(
         walker.walk(node)
         return True
 
-    def open(self) -> None:
-        """Initialize counters."""
+    def _set_astroid_options(self) -> None:
+        """Pass some config values to astroid.MANAGER object."""
         MANAGER.always_load_extensions = self.config.unsafe_load_any_extension
         MANAGER.max_inferable_values = self.config.limit_inference_results
         MANAGER.extension_package_whitelist.update(self.config.extension_pkg_allow_list)
         if self.config.extension_pkg_whitelist:
             MANAGER.extension_package_whitelist.update(
                 self.config.extension_pkg_whitelist
             )
-        self.stats.reset_message_count()
```

Comment:
In order to open checkers with per-file configs we need to call `_astroid_module_checker` per file => we need to call `_get_asts` before `_astroid_module_checker` => we need to save stats that are generated during `_get_asts`.

It seems that just not calling `reset_message_count()` has the same effect as something like `self.total_stats = merge_stats(self.total_stats, self.stats)` before `reset_message_count()`